VisualizationBase/src/nodes/InfoNode.cpp

```diff
@@ -52,18 +52,32 @@ InfoNode::InfoNode(Model::Node *target, QJsonArray enabledInfos)
 	Q_ASSERT(target);
 	for (auto item : enabledInfos)
 		setEnabled(item.toString(), true);
-	allInfoNodes.append(this);
+	initialize();
 }
 
 InfoNode::~InfoNode()
 {
-	allInfoNodes.removeAll(this);
+	allInfoNodes.remove(key_);
 	target_ = nullptr;
 }
 
+void InfoNode::initialize()
```

Comment:
So this all feels quite unnecessary. If you only really need `allInfoNodes`  just to be able to access a node from Javascript, then why not:
- remove `allInfoNodes` and `initialize`
- instead of some magical id, just pass a pointer to the node to javascript.
- the javascript can just pass this pointer back to functions that need to know about the node.
- In the worst case, you can convert this pointer to string, and back to a pointer again later. A bit of a hack, but still better than writing all this extra stuff.

Do you need `allInfoNodes` for anything but javascript?